bundles/org.openhab.binding.kodi/src/main/resources/OH-INF/thing/thing-types.xml

```diff
@@ -560,4 +562,16 @@
 		<description>Profile of Kodi</description>
 		<state pattern="%s"/>
 	</channel-type>
+	<channel-type id="screensaver">
+		<item-type>Switch</item-type>
+		<label>Screensaver</label>
+		<description>Status of the screensaver</description>
+		<state readOnly="true"/>
+	</channel-type>
+	<channel-type id="inputrequested">
```

Comment:
I propose to model this as a trigger channel which emits the requested input type (or "FINISHED"). This way a user can implement different reactions based on the requested type. Wdyt?

Reply:
I'm trying to get a change into Kodi which allows detecting if an Input is requested or not. Currently, if openHAB is started up after the input is requested it wouldn't notice it, even though the dialog is still open. The idea was to add a boolean for that, so that it could be implemented similar to the screensaver detection. If this would indeed be modeled as a trigger channel this future enhancement would be a lot harder.

Also we actually have "Title" (like "Enter search String") and the type, so ideally we would give access to both. If you still think it's better to make it a trigger channel (with the disadvantage that at least I won't be able to implement the polling on startup/reconnection) then it might make more sense to use the title and not the type.

Reply:
This will not work either. The `Input.OnInputRequested` is a Kodi notification which will be triggered only once when the user input is necessary. Thus if openHAB is not available at that specific moment or the websocket of the binding is not connected it will never be recognized by OH or set after a restart. You are not polling for it. Even more the state of the item linked to the channel will be `NULL` after an restart.

Regarding the title I can think of an OH enhancement to define additional properties delivered by an event. Currently an event can only have a type no other payload.

Reply:
Exactly, there is currently no way of polling it in Kodi. But I am working on implementing that, it will just be a boolean though.

Reply:
Alright. Please ping me once you got it implemented.

Reply:
Just to confirm: You want to wait for the change before merging? That change needs to be done at Kodi first and the maintainers there have already made clear on a different PR that there won't be API-changes between versions, so before it could actually be used a new version needs to be released with those changes in. I don't think waiting for that is a good idea as that could very well take another year or so, so I'd rather kick out this change if it's a reason not to merge and put it in a diferent PR.

Reply:
@cweitkamp, @Flole998 - can this discussion be concluded? Anything preventing this PR for being merged?

Reply:
Yes, I am fine with it. We can merge this now.